tools/integration_tests/list_large_dir/list_large_dir_test.go

```diff
@@ -0,0 +1,70 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provide test for listing large directory
+package list_large_dir_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const DirectoryWithTwelveThousandFiles = "directoryWithTwelveThousandFiles"
+const PrefixFileInDirectoryWithTwelveThousandFiles = "fileInDirectoryWithTwelveThousandFiles"
+const DirectoryWithTwelveThousandFilesAndHundredExplicitDir = "directoryWithTwelveThousandFilesAndHundredExplicitDir"
+const DirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = "directoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir"
+const PrefixFileInDirectoryWithTwelveThousandFilesAndHundredExplicitDir = "fileInDirectoryWithTwelveThousandFilesAndHundredExplicitDir"
+const NumberOfFilesInDirectoryWithTwelveThousandFiles = 12000
+const NumberOfFilesInDirectoryWithTwelveThousandFilesAndHundredExplicitDir = 12000
+const NumberOfFilesInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = 12000
+const NumberOfExplicitDirsInDirectoryWithTwelveThousandFilesAndHundredExplicitDir = 100
+const NumberOfImplicitDirsInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = 100
+const PrefixFileInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = "fileInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir"
+const NumberOfExplicitDirsInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = 100
+const ExplicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = "explicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir"
+const ImplicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir = "implicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDirAndHundredImplicitDir"
+const ExplicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDir = "explicitDirInDirectoryWithTwelveThousandFilesAndHundredExplicitDir"
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	// --o=ro flag is for stopping the deletion of objects before and after testing.
+	// https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/setup/setup.go#L172
+	flags := [][]string{{"--implicit-dirs", "--o=ro"}}
+
+	if setup.TestBucket() != "" && setup.MountedDirectory() != "" {
+		log.Printf("Both --testbucket and --mountedDirectory can't be specified at the same time.")
+		os.Exit(1)
+	}
+
+	// Run tests for mountedDirectory only if --mountedDirectory flag is set.
+	setup.RunTestsForMountedDirectoryFlag(m)
+
+	testBucket := setup.TestBucket()
+
+	setup.SetTestBucket("integration-test-data-gcsfuse")
```

Comment:
Around 15 to 20 mins for One Test.
If we are going to create three different directory for three tests- it will take 50 to 60mins.
We can may be append directory structure like first 12000 files , then 100 explicit dirs and then for third test 100 implicit dirs. In that case may be it will take 20 mins to pass all the tests.

Reply:
are GCE VM and bucket in same zone ?